### PR TITLE
enabled typescript eslint plugin strict

### PR DESCRIPTION
**Pull request by chwoerz**

### CODE REVIEW

**Comment by rainerhahnekamp** [comment]
Thanks a lot Christian. Will take a look at it as soon as I can.

**Comment by chwoerz** [comment]
Take your time :)

File: packages/core/src/lib/file-info/file-info.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class FileInfo {
   }
 }
 
-type NestedArray = Array<string | NestedArray>;
+type NestedArray = (string | NestedArray)[];
```
Comment (rainerhahnekamp) [inline packages/core/src/lib/file-info/file-info.ts:22]: What is the a reason why we go here with `Array`?

File: packages/core/src/lib/test/project-configurator.ts [hunk 2]
```diff
@@ -23,15 +23,18 @@ export type FileTreeContentType =
   | string[]
   | SheriffConfigContent;
 
-export type FileTree = { [key: string]: FileTreeContentType };
+// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
```
Comment (rainerhahnekamp) [inline packages/core/src/lib/test/project-configurator.ts:26]: Sticking to the type here would cause another linter error?

File: yarn.lock
Comment (rainerhahnekamp) [inline yarn.lock:1]: This one bothers me a bit. Why are there so many removals?

**Review by rainerhahnekamp — CHANGES_REQUESTED** [review]
@chwoerz I've left some comments. Maybe you can look over them and add some feedback. Thank you.

File: packages/core/src/lib/file-info/file-info.ts (hunk 1 above)
Comment (chwoerz) [reply packages/core/src/lib/file-info/file-info.ts:22]: EsLint has thrown an error

File: packages/core/src/lib/test/project-configurator.ts (hunk 2 above)
Comment (chwoerz) [reply packages/core/src/lib/test/project-configurator.ts:26]: Exactly. It will say it is a circular reference

File: yarn.lock
Comment (chwoerz) [reply yarn.lock:1]: interesting. i've updated the typescript-eslint plugin version. I will revert it and see if it sticks

File: packages/core/src/lib/test/project-configurator.ts (hunk 2 above)
Comment (rainerhahnekamp) [reply packages/core/src/lib/test/project-configurator.ts:26]: Okay, I'm not really happy with ignoring eslint errors, but maybe this is something we can cover in a follow-up issue

**Pull request merged**